### PR TITLE
Feat/migrate user account

### PR DESCRIPTION
**Pull request by varun257**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix #MANAGER-5820
| License          | BSD 3-Clause

## Description

Migration of user account to a standalone module

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report test#16720.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

File: packages/manager/apps/dedicated/client/app/account/otrs/otrs.controller.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ import isDate from 'lodash/isDate';
 import map from 'lodash/map';
 import set from 'lodash/set';
 
+import moment from 'moment';
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/account/otrs/otrs.controller.js:9]: `moment` is a global according to eslint and is imported with a script loader

File: packages/manager/apps/user-account/src/app.module.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    /* @ngInject */ ($locationProvider) => $locationProvider.hashPrefix(''),
+  )
+  .run(
+    /* @ngInject */ ($transitions) => {
+      const unregisterHook = $transitions.onSuccess({}, () => {
+        detachPreloader();
+        unregisterHook();
+      });
+    },
+  );
```
Comment (marie-j) [inline packages/manager/apps/user-account/src/app.module.js:46]: A redirect to default app url could be nice to avoid landing on a blank screen when starting the app

**Comment by ovh-cds** [comment]
CDS Report test#16763.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

**Comment by ovh-cds** [comment]
CDS Report test#16765.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

**Comment by ovh-cds** [comment]
CDS Report test#16805.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

File: packages/manager/apps/dedicated/client/app/account/otrs/otrs.controller.js (hunk 1 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/account/otrs/otrs.controller.js:9]: Have added it in `eslint` config.

**Comment by varun257** [comment]
Has to be merged after `refactor/dedicated-univers-components`

**Comment by ovh-cds** [comment]
CDS Report test#16811.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

File: packages/manager/apps/dedicated/package.json [hunk 3]
```diff
@@ -44,6 +44,7 @@
     "@ovh-ux/manager-preloader": "^1.0.0",
     "@ovh-ux/manager-server-sidebar": "^2.0.0 || ^3.0.0",
     "@ovh-ux/manager-support": "^0.6.0 || ^0.7.0 || ^1.0.0",
+    "@ovh-ux/manager-user-account": "^0.0.0",
```
Comment (marie-j) [inline packages/manager/apps/dedicated/package.json:47]:
```suggestion
    "@ovh-ux/manager-user-account": "^0.0.0 || ^1.0.0",
```
and module should be release with a breaking change commit

**Comment by ovh-cds** [comment]
CDS Report test#16863.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

File: packages/manager/apps/dedicated/client/app/account/otrs/otrs.app.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import ovhManagerCore from '@ovh-ux/manager-core';
+import angular from 'angular';
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/account/otrs/otrs.app.js:1]: why did you remove  import ovhManagerCore from '@ovh-ux/manager-core';

File: packages/manager/apps/dedicated/package.json [hunk 5]
```diff
@@ -44,6 +44,7 @@
     "@ovh-ux/manager-preloader": "^1.0.0",
     "@ovh-ux/manager-server-sidebar": "^2.0.0 || ^3.0.0",
     "@ovh-ux/manager-support": "^0.6.0 || ^0.7.0 || ^1.0.0",
+    "@ovh-ux/manager-user-account": "^0.0.0 || ^1.0.0",
```
Comment (radireddy) [inline packages/manager/apps/dedicated/package.json:47]: you need to add dedicated-universe-components

File: packages/manager/apps/user-account/src/index.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.user-account {
```
Comment (radireddy) [inline packages/manager/apps/user-account/src/index.scss:1]: its good idea to move these to module

File: packages/manager/modules/dedicated-universe-components/src/components/stepper/components/button/navigation-button.component.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import controller from './navigation-button.controller';
```
Comment (radireddy) [inline packages/manager/modules/dedicated-universe-components/src/components/stepper/components/button/navigation-button.component.js:1]: i have removed stepper from dedicated universe components, please rebase

File: packages/manager/modules/user-account/package.json [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    "punycode": "^1.4.1"
+  },
+  "peerDependencies": {
+    "@ovh-ux/manager-beta-preference": "^1.0.0",
+    "@ovh-ux/manager-config": "^1.0.0 || ^2.0.0",
+    "@ovh-ux/manager-core": "^10.0.0 || ^11.0.0",
+    "@ovh-ux/manager-models": "^1.0.0",
+    "@ovh-ux/ng-at-internet": "^5.3.0",
+    "@ovh-ux/ng-at-internet-ui-router-plugin": "^3.1.0",
+    "@ovh-ux/ng-ovh-dedicated-universe-components": "0.0.0",
```
Comment (radireddy) [inline packages/manager/modules/user-account/package.json:25]:
```suggestion
    "@ovh-ux/ng-ovh-dedicated-universe-components": "^0.0.0 | ^1.0.0",
```

File: packages/manager/modules/user-account/src/css/source.scss [hunk 9]
```diff
@@ -0,0 +1,8 @@
+$flag-icon-css-path: '~flag-icon-css/flags';
+
+@import 'flag-icon-css/sass/flag-icon';
+@import 'bootstrap4/scss/_functions.scss';
```
Comment (radireddy) [inline packages/manager/modules/user-account/src/css/source.scss:4]: you have imported bootstrap in app as well, do you need it in two places?

File: packages/manager/modules/user-account/src/user/advanced/user-advanced.controller.js [hunk 10]
```diff
@@ -1,14 +1,13 @@
 export default /* @ngInject */ function(
-  userAccountServiceInfos,
+  UserAccountServiceInfos,
```
Comment (radireddy) [inline packages/manager/modules/user-account/src/user/advanced/user-advanced.controller.js:2]: why did you change userAccountServiceInfos to UserAccountServiceInfos?iss it intentional?

**Review by radireddy — COMMENTED** [review]
@varun257 I did notice lazyloading for root account route. Please add it if not added.

File: packages/manager/apps/dedicated/client/app/account/otrs/otrs.app.js (hunk 4 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/account/otrs/otrs.app.js:1]: It has been removed by mistake, i have not touched otrs and contacts

File: packages/manager/modules/user-account/src/css/source.scss (hunk 9 above)
Comment (varun257) [reply packages/manager/modules/user-account/src/css/source.scss:4]: I have removed this in modules, retained in apps

File: packages/manager/apps/user-account/src/index.scss (hunk 6 above)
Comment (varun257) [reply packages/manager/apps/user-account/src/index.scss:1]: I need this in app as bootstrap is already included in dedicated

File: packages/manager/modules/user-account/src/user/advanced/user-advanced.controller.js (hunk 10 above)
Comment (varun257) [reply packages/manager/modules/user-account/src/user/advanced/user-advanced.controller.js:2]: Yes it is. The class name while exporting is `userAccountServiceInfos` and actual service itself is `UserAccountServiceInfos`

File: packages/manager/apps/user-account/src/index.scss (hunk 6 above)
Comment (varun257) [reply packages/manager/apps/user-account/src/index.scss:1]: Has been retained in `apps` as it is only required for `standalone` module

**Comment by ovh-cds** [comment]
CDS Report test#16886.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

**Comment by JeremyDec** [comment]
Hello @varun257 , there's conflicts on translations files

**Comment by ovh-cds** [comment]
CDS Report test#17061.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✘

**Comment by varun257** [comment]
No longer required, a new PR is created targeting a different branch
